feat/level_1 タスクを新しく追加する機能の実装

nakatomoyanyan opened the pull request:
・作業時間　3日間合計7h
・追加した機能の概要
タスク管理アプリの画面のレイアウト、追加ボタンを押すとタスクを追加する機能
・実装した機能の指示内容
https://qiita.com/foolproof-htb/private/73ad2fea8c27512fbf12
・タスク管理アプリのレイアウト
![スクリーンショット 2024-06-24 14 27 07](https://github.com/nakatomoyanyan/frontend_practice/assets/173334729/7b8e8c2a-ef8a-437a-bc90-581d6591fcda)
・タスク追加時の動作のgif
![level1-4](https://github.com/nakatomoyanyan/frontend_practice/assets/173334729/b6137105-7b2f-4238-83e5-735c3ce9de8c)

nakatomoyanyan committed:
aboutページ実装

aboutページを実装しました。

nakatomoyanyan committed:
入力ボタン及び選択ボタン、追加ボタンの実装

入力ボタン、選択ボタン、追加ボタンを実装しました。

nakatomoyanyan committed:
追加ボタンクリック時の動作実装

追加ボタンクリック時の動作を実装しました。
入力されたタスクの名前、期日、優先度をmain下部に表示します。
タスクの名前の長さに応じてタスクの大きさが変わるようにしました。

nakatomoyanyan committed:
入力フォームのデフォルトの値を設定

入力フォームのデフォルトの値を設定しました。

nakatomoyanyan committed:
タイトルの大きさ変更

タイトルの大きさを30pxから40pxにしました。

foolproof-htb commented on index.html:
```diff
@@ -11,20 +11,32 @@
                 <ul>
                     <a href="#" class="home">ホーム</a>
-                    <a href="#" class="about-application">このアプリについて</a>
+                    <a href="http://127.0.0.1:5500/about.html" class="about-application">このアプリについて</a>
```
ローカルホストを参照するようになってます！

foolproof-htb commented on task_add.js:
```diff
@@ -0,0 +1,34 @@
+    //タスクのフォーム作成
+    let taskList = document.querySelector('.task_list');
+    let newTask = document.createElement('div');
+    newTask.style.border = 'solid #cccccc';
+    newTask.style.borderRadius = '5px';
+    newTask.style.display = 'inline-block';//タスク名の長さに応じてタスクのフォームの大きさが変わる
```
それぞれの役割をちゃんと分担させるためにも、スタイルは基本的にCSSでまとめて定義しておくと良いです。
なので、スタイルはCSSで定義した上でJSではclassを付与するような形にするとスマートかと思います！

foolproof-htb replied:
タスクの幅が可変である必要はないので、適当な幅を指定すれば inline-block にしなくて良くなりそうです。
また、そうすれば16〜17行目にあるような構造上不要な空のpタグも入れなくて済むかと思います。

foolproof-htb commented on task_add.js:
```diff
@@ -0,0 +1,34 @@
+    let nullPtag = document.createElement('p');//タスクのフォームを縦並びにする&フォーム同士に空白を開けるため
+    taskList.appendChild(nullPtag);
+    //タスク名を追加
+    let newTaskValue = document.createElement('p');
+    newTaskValue.innerText = taskValue.value;
+    newTask.appendChild(newTaskValue);
+    //タスクの期日を追加
+    let newTaskDate = document.createElement('p');
+    newTaskDate.innerText = '期限日:' + taskDate.value;
+    newTask.appendChild(newTaskDate);
+    //タスクの優先度を追加
+    let newTaskImportance = document.createElement('p');
+    newTaskImportance.innerText = '優先度:' + taskImportance.value;
+    newTask.appendChild(newTaskImportance);
```
1行ずつ追加する方法もありますが、以下のようにまとめて追加する方法もあります 👀 
```js
newTask.innerHTML = `
  <p>${taskValue.value}</p>
  <p>期限日:${taskDate.value}</p>
  <p>優先度:${taskImportance.value}</p>
`
```

foolproof-htb reviewed (approved):
まずは課題の実装 お疲れ様です！
何点かコメントしましたので、ご確認いただければと思います 🙇

nakatomoyanyan committed:
aboutページのリンクの修正

変更前：aboutページがローカルホストを参照するようになっていた。
変更後：aboutページのhtmlファイルを参照するように変更。

nakatomoyanyan committed:
cssの装飾をcssファイルにまとめた＆inline-blockの属性の削除

cssの指定はcssファイルのみにまとめました。
javascriptでは新たに「newtask」のクラス名を指定しました。
inline-blockの属性を消去しました。属性の消去に合わせて空のpタグを削除しました。
タスク間の間隔を調整するためにmargin-bottomを指定しました。

nakatomoyanyan committed:
タスクの中身(タスク名、期日、優先度)をまとめて追加

inner.HTMLのメソッドを用いてタスクの中身をまとめて追加するように書き換えました。

iwaguma-manabi commented on task_add.js:
```diff
@@ -0,0 +1,24 @@
+//追加ボタンをクリックした時の動作(タスク一覧にタスクを追加)
+taskSubmit.addEventListener('click',() => {
+    //タスクのフォーム作成
+    let taskList = document.querySelector('.task_list');
```
ここはクリック時の挙動として毎回必要な内容でしょうか？

iwaguma-manabi commented on task_add.js:
```diff
@@ -0,0 +1,24 @@
+
+//追加ボタンをクリックした時の動作(タスク一覧にタスクを追加)
+taskSubmit.addEventListener('click',() => {
+    //タスクのフォーム作成
```
作っているのはフォームではなくタスクではないかなと思いまいした

foolproof-htb reviewed (commented):
以下の対応は不要です！
一般的には、コードを見て内容が分かるような箇所にはコメントは不要なので、このコードをあとから読む第三者を信頼してコードだけ書いてもいいかと思います👍

nakatomoyanyan committed:
tasklistの要素の取得タイミング変更

変更前：追加ボタンクリック時
変更後：javascript読み込み時

SaitoJP commented on index.html:
```diff
@@ -11,20 +11,32 @@
                 <h1>タスクを追加する</h1>
+                <form>
+                    <input class="task_value" placeholder="タスク名" type="text"/>
+                    <input class="task_date" type="date" value="年/月/日" />
```
以下のドキュメントによるとvalueは日付のデータ（yyy-mm-dd）を投入する箇所ではないですかね？
https://developer.mozilla.org/ja/docs/Web/HTML/Element/input/date

nakatomoyanyan committed:
日付入力フォームのvalueの削除

日付入力フォームの不要であったvalueを削除しました。
